### DISCUSSION TITLE
Missing realtime data

### DISCUSSION THREAD

**Discussion by giligone** [ropensci/tidyhydat#173] [General]
I would first like to thank you for your R package “tidyhydat”. It is great and I use it regularly. I had a couple questions 
 
First, I’m trying to extract data for a site ([08NM158](https://wateroffice.ec.gc.ca/report/real_time_e.html?stn=08NM158)). The data are clearly up to date via the Water Office website (see graph).
 
 
![water office graph](https://user-images.githubusercontent.com/47872403/173957636-d472a46e-aa86-4dd6-813e-16602e75656c.png)

And I can see using tidyhydat that it has found the site and shows for “REAL_TIME” that it is TRUE
 

![station_search](https://user-images.githubusercontent.com/47872403/173957672-9c2d116f-1456-4b75-8d47-638c676a24f1.png)

 
But if I try and query these data I get this error
 
![realtime results](https://user-images.githubusercontent.com/47872403/173957692-38f8ba80-e8b0-4405-b169-6c597bc9154e.png)

When I do a query with hy_daily_flows I can see potentially the problem. This site was formerly active, discontinued in 1982, and then reactivated but the tidyhydat appears only to have the old data.
 
![hydaily](https://user-images.githubusercontent.com/47872403/173957708-560a3dbd-06c7-40e1-acdf-cff6c4de7010.png)

Is this likely to be the issue? Is there any way to work around this?
 
 
My second question is that this site also stores turbidity data. I know tidyhydat does not currently support this type of data but is there any potential that it could become supported in the future?
 
Thanks for your time 
Kind regards

**Comment by boshek** [comment]
So I am definitely not an authoritative voice for WSC. Should be clear on that. 

And really I don't have much of answer for you. The data from `hy_` functions all good from hydat. That is validated WSC data. The data from the datamart (i.e `realtime_*`) is from this site: https://dd.weather.gc.ca/hydrometric/csv/BC/

The datamart only offers the last 30 days of records and then there is a gap between that and when the data appears in HYDAT. I am not certain why this data, however, is _not_ in the datamart. Sorry I can't be more help!

**Reply by giligone** [reply]
Thanks for your reply. I'll try my luck inquiring directly to WSC to see if I can find anything. I'll post a follow-up if I learn anything. Thanks again for your help.

**Comment by boshek** [comment]
I should also mention that there is some suspended sediment data in HYDAT. They can be access via tidyhydat. Most of it has been discontinued but worth mentioning: 

``` r
library(tidyhydat)
hy_sed_daily_suscon("08MF005")
#>   Queried from version of HYDAT released on 2022-04-18
#>    Observations:                      5,358
#>    Measurement flags:                 2,751
#>    Parameter(s):                      Suscon
#>    Date range:                        1965-05-01 to 1979-12-31 
#>    Station(s) returned:               1
#>    Stations requested but not returned: 
#>     All stations returned.
#> # A tibble: 5,358 × 5
#>    STATION_NUMBER Date       Parameter Value Symbol
#>    <chr>          <date>     <chr>     <dbl> <chr> 
#>  1 08MF005        1965-05-01 Suscon       NA <NA>  
#>  2 08MF005        1965-05-02 Suscon       NA <NA>  
#>  3 08MF005        1965-05-03 Suscon       NA <NA>  
#>  4 08MF005        1965-05-04 Suscon       NA <NA>  
#>  5 08MF005        1965-05-05 Suscon       NA <NA>  
#>  6 08MF005        1965-05-06 Suscon       NA <NA>  
#>  7 08MF005        1965-05-07 Suscon       NA <NA>  
#>  8 08MF005        1965-05-08 Suscon       NA <NA>  
#>  9 08MF005        1965-05-09 Suscon       NA <NA>  
#> 10 08MF005        1965-05-10 Suscon       NA <NA>  
#> # … with 5,348 more rows
```

**Comment by hydropatch** [comment]
Hi Sam and giligone 
I contacted ECCC and they have now made that station available on the datamart so realtime_dd(station_number = "08NM158") now works!

**Reply by boshek** [reply]
Hey! That's great. Curious @patjlittle - which email did you contact them at?

**Reply by giligone** [reply]
Brilliant! Thanks @patjlittle. I tried contacting them too but had no success so I appreciate your efforts. I can also confirm that it is working for me too.
